src/Compilers/CSharp/Portable/Symbols/MemberSignatureComparer.cs

```diff
@@ -140,20 +153,22 @@ internal sealed class MemberSignatureComparer : IEqualityComparer<Symbol>
             considerCallingConvention: false,
             considerRefKindDifferences: true,
             considerArity: false,
-            typeComparison: TypeCompareKind.ObliviousNullableModifierMatchesAny);
+            typeComparison: TypeCompareKind.AllIgnoreOptions);
 
         /// <summary>
-        /// This instance is used to determine if a partial method implementation matches the definition,
-        /// including differences ignored by the runtime.
+        /// Determines if an interceptor has a compatible signature with an interceptable method.
+        /// If methods are considered equal by <see cref="InterceptorsComparer"/>, but not equal by this comparer, a warning is reported.
+        /// NB: when a classic extension method is intercepting an instance method call, a normalization to 'ReducedExtensionMethodSymbol' must be performed first.
         /// </summary>
-        public static readonly MemberSignatureComparer PartialMethodsStrictComparer = new MemberSignatureComparer(
-            considerName: true,
-            considerExplicitlyImplementedInterfaces: true,
+        public static readonly MemberSignatureComparer InterceptorsStrictComparer = new MemberSignatureComparer(
+            considerName: false,
+            considerExplicitlyImplementedInterfaces: false,
             considerReturnType: true,
             considerTypeConstraints: false,
             considerCallingConvention: false,
             considerRefKindDifferences: true,
-            typeComparison: TypeCompareKind.ObliviousNullableModifierMatchesAny);
+            considerArity: false,
+            typeComparison: TypeCompareKind.AllNullableIgnoreOptions | TypeCompareKind.IgnoreTupleNames);
```

Comment:
>AllNullableIgnoreOptions [](http://example.com/codeflow?start=44&length=24)

Why ignore nullable differences rather than using `ObliviousNullableModifierMatchesAny`?

Reply:
We are checking for unsafe nullability differences separately. I decided to just not report on *safe* nullability differences.

When nullable differences are detected by signatures comparing not-equal like this it's a pain, because the differences might be `dynamic`/`object` stuff, or it might be `string?`/`string` stuff, which we might want to group separately, so that `#nullable disable warnings` disables the latter and not the former.

This is why the following "bug" occurs for partial methods.
```cs
#nullable enable annotations
#nullable disable warnings

public partial class C
{
    // implementation accepting "less" than definition is unsafe, and we don't warn here with '#nullable disable warnings'
    public partial void M1(string? s);
    public partial void M1(string s) { }

    // implementation accepting "more" than definition is safe, but we warn here with '#nullable disable warnings'
    public partial void M2(string s);
    public partial void M2(string? s) { }
}
```

Reply:
Discussed a bit offline. At this point, permitting "safe" variance without warnings seems like the right thing for nullability and scoped. I feel like erroring on `object`/`dynamic` differences is right here and ignoring tuple name differences is right. It feels a tiny bit arbitrary sometimes...but open to feedback on some of these decisions.